# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/shared/LIR-shared.h
# Commit: 90fa382b5a20
# Full Hash: 90fa382b5a203d2e28bfa69f29ab8edee150b77b
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2021-12-23 09:34:36
# Regressor Bug: 1642412
# File Overlap Count: 8
# Description:
#   Bug 1642412 - wasm: Optimize ion table.get/set/size. r=lth
#   
#   This commit adds ion fast paths for table.get/set when the table
#   representation is anyref. An unconditional fast path is added for
#   table.size.
# ==============================================================================

diff -r 40f861f62a0f -r 90fa382b5a20 js/src/jit/shared/LIR-shared.h
--- a/js/src/jit/shared/LIR-shared.h	Wed Dec 22 17:36:29 2021 +0000
+++ b/js/src/jit/shared/LIR-shared.h	Wed Dec 22 17:36:30 2021 +0000
@@ -3156,6 +3156,20 @@
   size_t offset() { return mirRaw()->toWasmDerivedPointer()->offset(); }
 };
 
+class LWasmDerivedIndexPointer : public LInstructionHelper<1, 2, 0> {
+ public:
+  LIR_HEADER(WasmDerivedIndexPointer);
+  explicit LWasmDerivedIndexPointer(const LAllocation& base,
+                                    const LAllocation& index)
+      : LInstructionHelper(classOpcode) {
+    setOperand(0, base);
+    setOperand(1, index);
+  }
+  const LAllocation* base() { return getOperand(0); }
+  const LAllocation* index() { return getOperand(1); }
+  Scale scale() { return mirRaw()->toWasmDerivedIndexPointer()->scale(); }
+};
+
 class LWasmParameterI64 : public LInstructionHelper<INT64_PIECES, 0, 0> {
  public:
   LIR_HEADER(WasmParameterI64);